### DISCUSSION TITLE
2022-04-29: 20 Million PyPI Downloads

### DISCUSSION THREAD

**Discussion by JorjMcKie** [pymupdf/PyMuPDF#1595] [Announcements]
Since its presence on [PyPI](https://pypi.org/project/PyMuPDF/) around midyear 2016, PyMuPDF has been downloaded more than ~13 (2022-02)~ ~17 (2022-03)~ **20 million times** (2022-04-29).

The **_two million_** mark was reached at the end of February 2021. At end of February 2022 we had 13 millions.
Which means that PyMuPDF has been downloaded **_11 million times_** within one year!

**Comment by qwertynik** [comment]
That's great news @JorjMcKie. 
With the amount of care 🤝 and attention 🔎 you provide to this library, also reflected by your participation in the Discussions, this is a _natural_ outcome.

**Reply by JorjMcKie** [reply]
Thank you very much for your appreciation.

**Comment by JorjMcKie** [comment]
The download numbers continue to increase rapidly. Looks like we will exceed 13 million total downloads by tomorrow 23rd ...

**Reply by qwertynik** [reply]
Great news @JorjMcKie and interesting too. What do you think is spiking the downloads? Does it have something to do with some new upgrades/innovations in the PDF file format? Of course, there's this pandemic that has changed many things.

**Reply by JorjMcKie** [reply]
I am wondering myself. I did not really introduce groundbreaking features recently.
Yes, there was this finally using MuPDF's OCR feature, the rest was only little things.
But maybe it is like with an infection itself: the more people use it, the greater is the number who talk about it and make others give it a try. Or maybe a snowball is a better metaphor: it may lead to an avalanche ...

But thank you for watching it and being so interested!

**Reply by JorjMcKie** [reply]
BTW: we **_are_** at 13 million.
This the download curve: interest is massively increasing since end of last year.

![Picture1](https://user-images.githubusercontent.com/8290722/155298033-72aa9f26-c548-4df3-9dd8-9bc2be19c7d0.png)

**Reply by qwertynik** [reply]
> But thank you for watching it and being so interested!

Appreciation is the _least_ that everyone should have when using an open-source project.
Good the see the project gain traction.

**Comment by ousia** [comment]
@JorjMcKie, congratulations for your excellent work.

You really deserve the recognition of such a huge amount of downloads.

**Reply by JorjMcKie** [reply]
Hey Pablo, thank you very much for your nice comment! As per today, it looks like even 20 millions within the next 5 days.

**Reply by sunov** [reply]
@JorjMcKie , Congratulations for your great work
